tests/Tests.FeatureManagement.AspNetCore/FeatureGateEndpoint.cs

```diff
@@ -0,0 +1,181 @@
+        {
+            var context = new TargetingContext
+            {
+                UserId = _userId,
+                Groups = _groups
+            };
+            return new ValueTask<TargetingContext>(context);
+        }
+    }
+
+    public class FeatureTestServer : IDisposable
+    {
+        private readonly IHost _host;
+        private readonly HttpClient _client;
+        private readonly IDictionary<string, string> _featureSettings;
+        private readonly ITargetingContextAccessor _targetingContextAccessor;
+
+        public FeatureTestServer(
+            IDictionary<string, string> featureSettings = null,
+            ITargetingContextAccessor targetingContextAccessor = null)
+        {
+            _featureSettings = featureSettings ?? new Dictionary<string, string>
+            {
+                ["FeatureManagement:TestFeature"] = "true"
+            };
+            _targetingContextAccessor = targetingContextAccessor ?? new TestTargetingContextAccessor();
+            _host = CreateHostBuilder().Build();
+            _host.Start();
+            _client = _host.GetTestServer().CreateClient();
+        }
+
+        private IHostBuilder CreateHostBuilder()
+        {
+            var configuration = new ConfigurationBuilder()
+                .AddInMemoryCollection(_featureSettings)
+                .Build();
+
+            return Host.CreateDefaultBuilder()
+                .ConfigureWebHost(webBuilder =>
+                {
+                    webBuilder
+                        .UseTestServer()
+                        .ConfigureServices(services =>
+                        {
+                            services.AddSingleton<IConfiguration>(configuration);
+                            services.AddSingleton(_targetingContextAccessor);
+                            services.AddFeatureManagement()
+                                    .AddFeatureFilter<TargetingFilter>();
+                            services.AddRouting();
+                        })
+                        .Configure(app =>
+                        {
+                            app.UseRouting();
+                            app.UseMiddleware<TargetingHttpContextMiddleware>();
+                            app.UseEndpoints(endpoints =>
+                            {
+                                endpoints.MapGet("/test", new Func<string>(() => "Feature Enabled"))
+                                    .WithFeatureGate("TestFeature");
+                                endpoints.MapGet("/test-targeting", new Func<string>(() => "Feature With Targeting Enabled"))
+                                    .WithFeatureGate("TestFeatureWithTargeting");
+                            });
+                        });
+                });
+        }
+
+        public HttpClient Client => _client;
+
+        public void Dispose()
+        {
+            _host?.Dispose();
+            _client?.Dispose();
+        }
+    }
+
+    public class FeatureGateEndpointFilterTests
```

Comment:
I expect this to just work, but I think adding a test for an endpoint group (`MapGroup`) can be useful.

For example, that the endpoints within a group are protected when a feature gate is added to the group.